### PR TITLE
Initial commit of Redash chart.

### PR DESCRIPTION
**Pull request by grugnog**
**What this PR does / why we need it**:

This adds a [Redash](https://github.com/getredash/redash/) chart.

Redash is an open source collaborative data analytics and visualization tool.

**Special notes for your reviewer**:

A version of this chart has been running without issue in an internal context for several weeks.

Note that Redash is developed by a startup of the same name. I am not affiliated with that company, but I will open an issue on their Github tracker to make them aware of this PR and see if they have input or interest in co-maintaining.

### CODE REVIEW

**Comment by unguiculus** [comment]
Please compare with what you get with `helm create` using Helm 2.8+ and update accordingly.

**Comment by grugnog** [comment]
@unguiculus I think this should be ready for further review.

**Comment by grugnog** [comment]
@unguiculus did you get a chance to look at the revisions?

File: incubator/redash/templates/_helpers.tpl
Comment (unguiculus) [inline incubator/redash/templates/_helpers.tpl]: If `fullnameOverride` is set, you get the same for `redash.fullname`, `redash.adhocWorker.fullname`, etc. I guess this is not intended and will cause problems/name clashes.

File: incubator/redash/templates/adhocworker-deployment.yaml
Comment (unguiculus) [inline incubator/redash/templates/adhocworker-deployment.yaml]: apps/v1beta2

File: incubator/redash/templates/scheduledworker-deployment..yaml
Comment (unguiculus) [inline incubator/redash/templates/scheduledworker-deployment..yaml]: apps/v1beta2

File: incubator/redash/templates/server-deployment.yaml
Comment (unguiculus) [inline incubator/redash/templates/server-deployment.yaml]: apps/v1beta2

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: incubator/redash/templates/_helpers.tpl
Comment (unguiculus) [inline incubator/redash/templates/_helpers.tpl]: Do you really need separate fullnameOverride for the components and configurable component names? How about doing something like this:

```
{{- define "redash.adhocWorker.fullname" -}}
{{- template "redash.fullname" . -}}-adhoc-worker
{{- end -}}
```

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: incubator/redash/templates/_helpers.tpl
Comment (grugnog) [reply incubator/redash/templates/_helpers.tpl]: @unguiculus I don't need this personally - I was trying mainly to keep things consistent with the Helm 2.8+ [create template](https://github.com/kubernetes/helm/blob/master/pkg/chartutil/create.go#L272) and also copying the pattern used by some of the other multi-deployment charts like [Prometheus](https://github.com/kubernetes/charts/blob/master/stable/prometheus/templates/_helpers.tpl).

If you think it's cleaner to drop fullnameOverride I am happy to remove it. If so let me know if you think fullnameOverride should be removed from all deployments or kept on the primary/web deployment.

Comment (unguiculus) [reply incubator/redash/templates/_helpers.tpl]: I'd suggest you follow the example above. You'll have one fullnameOverride affecting all derived fullnames.

**Comment by jurgenweber** [comment]
can we get this finished and merged.. It would be fabulous.

**Comment by grugnog** [comment]
@unguiculus I think this should be ready for further review.

File: incubator/redash/values.yaml
Comment (unguiculus) [inline incubator/redash/values.yaml]: Don't set a default value for the password

File: incubator/redash/values.yaml
Comment (unguiculus) [inline incubator/redash/values.yaml]: Don't set a default value for the password

**Review by unguiculus — CHANGES_REQUESTED** [review]
I'd suggest you move the chart to stable before we merge.

File: incubator/redash/values.yaml
Comment (drslump) [inline incubator/redash/values.yaml]: Is there a way to define an external postgresql database instead of running its own? for setups that already have a DB running (i.e. using AWS managed service) this would remove the need for persistence too.

**Comment by analyticalmonk** [comment]
@grugnog @unguiculus Hi, are there plans to merge this anytime soon? It would be great to have a redash helm chart available in the stable repo.

**Comment by thangbn** [comment]
this is really helpful, I've been using this for some time now. it's pretty stable

**Comment by sfrique** [comment]
Hey, i think redash plans to release the v5 sometime in the near future.
As this charts wasn't merge or anything like this, wouldn't make sense to start the work and get v5 out instead of v4 and then upgrade to v5? Just a thought. 

I hope to be using the v5 on kubernetes when it's released. I have an working v4 not on kubernetes, was about to migrate to kubernetes, but then i saw something about the update so I am going to wait for it.

I am willing to do some work on the v5 for kubernetes if needed!

**Comment by arikfr** [comment]
>  As this charts wasn't merge or anything like this, wouldn't make sense to start the work and get v5 out instead of v4 and then upgrade to v5? Just a thought.

I don't think there should be anything in this Chart that should change (aside from the Docker image reference) between v4 and v5. Hence no point in waiting for v5 to finalize or merge this.

**Comment by sfrique** [comment]
@arikfr great!!

I didn't take a look on redash v5 upgrade neither this chart yet, so i wouldn't know, i just added my 2c!

**Comment by grugnog** [comment]
Sorry for the delay in getting back to this. I have moved to stable and also updated the image tag.

Avoiding default passwords is a bit complex as Redash expects the password as part of connection URL strings (REDASH_DATABASE_URL and REDIS_URL). The autogenerated passwords are in a secret and not accessible during templating, so we would need to inject a script via Configmap to assemble each URL environment variable as part of the container startup from the parts (hostname, password etc). Alternatively we could see if Redash would be able to update the Docker image to accept the URL components instead of just a single URL string?

**Comment by grugnog** [comment]
I am working on a change to split out DB env variables and merge them in a shell script loader (that could hopefully eventually become part of the upstream entrypoint). This should allow Redash to be configured to use the autogenerated. It's still a WIP, but at https://github.com/grugnog/charts/commits/redash-env if you want to take a look at the direction.

**Comment by Guyzeroth** [comment]
@grugnog Got this running today and it seems fine. Would be cool to see it merged :) .

**Comment by ricktbaker** [comment]
Don't see an option to use an external postgres db.   Is that going to be possible with this chart?

**Comment by pachecobruno** [comment]
Hello! @grugnog this Chart is a great idea! Thank you!!

I'm having some issues using it. See if you can help me? I'm running with default values.`

1) After UP the application, the URL returns "Internal Server Error". The log is showing:
```
ProgrammingError: (psycopg2.ProgrammingError) relation "organizations" does not exist
```
I believe that redash was not able to create the Postgres tables.

2) When applying the command to create the tables manually. Just to poke and see what would happen:

```
$ bin / run ./manage.py database create_tables
sed: can not read .env: No such file or directory
[2018-12-08 13: 27: 48,823] [PID: 62] [INFO] [root] Generating grammar tables from /usr/lib/python2.7/lib2to3/Grammar.txt
[2018-12-08 13: 27: 48,846] [PID: 62] [INFO] [root] Generating grammar tables from /usr/lib/python2.7/lib2to3/PatternGrammar.txt
Traceback (most recent call last):
  File "/app/manage.py", line 6, in <module>
    from redash.cli import manager
  File "/app/redash/__init__.py", line 76, in <module>
    reset_new_version_status ()
  File "/app/redash/version_check.py", line 34, in reset_new_version_status
    latest_version = get_latest_version ()
  File "/app/redash/version_check.py", line 40, in get_latest_version
    return redis_connection.get (REDIS_KEY)
  File "/usr/local/lib/python2.7/dist-packages/redis/client.py", line 880, in get
    return self.execute_command ('GET', name)
  File "/usr/local/lib/python2.7/dist-packages/redis/client.py", line 578, in execute_command
    connection.send_command (* args)
  File "/usr/local/lib/python2.7/dist-packages/redis/connection.py", line 563, in send_command
    self.send_packed_command (self.pack_command (* args))
  File "/usr/local/lib/python2.7/dist-packages/redis/connection.py", line 538, in send_packed_command
    self.connect ()
  File "/usr/local/lib/python2.7/dist-packages/redis/connection.py", line 442, in connect
    raise ConnectionError (self._error_message (e))
redis.exceptions.ConnectionError: Error 111 connecting to localhost: 6379. Connection refused.
```

3) A observation: in my cluster the POD for the redash server takes approximately 10min to get the Running state. This is normal?

**Comment by ekimia** [comment]
excited! Let me know if you need any help @grugnog

**Comment by okgolove** [comment]
Any updates here?

**Comment by grugnog** [comment]
@unguiculus _Finally_ found some time to get the database schema creation working nicely with automatic password generation.

I also upgraded Redash and dependency chart versions, added an external PostgreSQL option and rebased to clean up the commits slightly.

Testing and review welcome from all!

File: stable/redash/Chart.yaml
Comment (unguiculus) [inline stable/redash/Chart.yaml]: Go for a `1.0.0`

File: stable/redash/README.md
Comment (unguiculus) [inline stable/redash/README.md]: Probably 1.9+

File: stable/redash/templates/adhocworker-deployment.yaml
Comment (unguiculus) [inline stable/redash/templates/adhocworker-deployment.yaml]: `apps/v1`

File: stable/redash/templates/scheduledworker-deployment.yaml
Comment (unguiculus) [inline stable/redash/templates/scheduledworker-deployment.yaml]: `apps/v1`

File: stable/redash/templates/server-deployment.yaml
Comment (unguiculus) [inline stable/redash/templates/server-deployment.yaml]: `apps/v1`

File: stable/redash/README.md
Comment (unguiculus) [inline stable/redash/README.md]: Remove line.

**Review by unguiculus — CHANGES_REQUESTED** [review]
Please update to use the new Kubernetes label conventions.
See https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md#names-and-labels

**Review by tomaytotomato — APPROVED** [review]

**Comment by tomaytotomato** [comment]
Having issues trying to start this branch with redash, 

```
RESOURCES:
==> v1/ConfigMap
NAME                          DATA  AGE
imprecise-lemur-redis         3     1s
imprecise-lemur-redis-health  3     1s
imprecise-lemur-redash        2     1s

==> v1/Service
NAME                                 TYPE       CLUSTER-IP      EXTERNAL-IP  PORT(S)   AGE
imprecise-lemur-postgresql-headless  ClusterIP  None            <none>       5432/TCP  1s
imprecise-lemur-postgresql           ClusterIP  10.152.183.211  <none>       5432/TCP  1s
imprecise-lemur-redis-master         ClusterIP  10.152.183.229  <none>       6379/TCP  1s
imprecise-lemur-redash               ClusterIP  10.152.183.111  <none>       80/TCP    1s

==> v1beta2/Deployment
NAME                                    DESIRED  CURRENT  UP-TO-DATE  AVAILABLE  AGE
imprecise-lemur-redash-adhocworker      1        1        1           0          1s
imprecise-lemur-redash-scheduledworker  1        1        1           0          1s
imprecise-lemur-redash                  1        1        1           0          1s

==> v1beta2/StatefulSet
NAME                          DESIRED  CURRENT  AGE
imprecise-lemur-postgresql    1        1        1s
imprecise-lemur-redis-master  1        1        1s

==> v1/Pod(related)
NAME                                                     READY  STATUS             RESTARTS  AGE
imprecise-lemur-redash-adhocworker-747f99bcbf-sn48v      0/1    ContainerCreating  0         1s
imprecise-lemur-redash-scheduledworker-5fdb4f49fc-w6vgg  0/1    ContainerCreating  0         1s
imprecise-lemur-redash-6db574f45c-kzkmx                  0/1    Init:0/1           0         1s
imprecise-lemur-postgresql-0                             0/1    Pending            0         1s
imprecise-lemur-redis-master-0                           0/1    Pending            0         1s

==> v1/Secret
NAME                        TYPE    DATA  AGE
imprecise-lemur-postgresql  Opaque  1     1s
imprecise-lemur-redis       Opaque  1     1s
imprecise-lemur-redash      Opaque  1     1s


NOTES:
1. Get the application URL by running these commands:
  export POD_NAME=$(kubectl get pods --namespace default -l "app=redash,release=imprecise-lemur" -o jsonpath="{.items[0].metadata.name}")
  echo "Visit http://127.0.0.1:8080 to use your application"
  kubectl port-forward $POD_NAME 8080:5000

```

It appears the redash workers keep crashing and restarting

```
imprecise-lemur-postgresql-0                              0/1     Pending            0          2m59s
imprecise-lemur-redash-6db574f45c-kzkmx                   0/1     Init:0/1           1          2m59s
imprecise-lemur-redash-adhocworker-747f99bcbf-sn48v       0/1     CrashLoopBackOff   4          2m59s
imprecise-lemur-redash-scheduledworker-5fdb4f49fc-w6vgg   0/1     CrashLoopBackOff   4          2m59s
imprecise-lemur-redis-master-0                            0/1     Pending            0          2m59s

```

**Helm Version** - 

```
Client: &version.Version{SemVer:"v2.12.3", GitCommit:"eecf22f77df5f65c823aacd2dbd30ae6c65f186e", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.12.3", GitCommit:"eecf22f77df5f65c823aacd2dbd30ae6c65f186e", GitTreeState:"clean"}

```

Any ideas?

File: stable/redash/Chart.yaml
Comment (mewwts) [inline stable/redash/Chart.yaml]: Should we target the recent version 6?

**Comment by grugnog** [comment]
@unguiculus @mewwts - thanks for the review - this should include all the changes requested.

@tomaytotomato I am testing successfully with Helm v2.12.3 on both Minikube v0.33.0 with k8s v1.13.0 and GKE v1.10.11. The issue seems to be that your postgres and redis are in "pending" state, perhaps waiting for a persistent volume claim? The other behaviour you are seeing is expected: main app deployment init container will block (then crash after some retries) and loop until those are both available, and the 2 worker deployments will crash loop until the main deployment init completes successfully.

**Comment by chriszs** [comment]
Have a Redash deployment running on an earlier version of this chart, was just able to update it to use this version. lgtm

**Comment by chriszs** [comment]
Update: May have spoke too soon. Having some problems with readiness and liveness probes. Will try to get to the bottom of that.

**Comment by chriszs** [comment]
Was getting readiness and liveness probe failures. Commented out readiness probe, still got this liveness probe failure:
```
Liveness probe failed: [...] File "/usr/local/lib/python2.7/dist-packages/redis/connection.py", line 442, in connect raise ConnectionError(self._error_message(e)) redis.exceptions.ConnectionError: Error 111 connecting to localhost:6379. Connection refused.
```

Because of "localhost," suspected env variable for Redis hostname wasn't getting passed to the liveness probe command. Tried swapping it out for:
```
          livenessProbe:
            exec:
              command: ["/bin/sh", "-c", ". /config/dynamicenv.sh && /app/manage.py status"]
```

Seemed to work. At least, not crashing anymore.

**Comment by grugnog** [comment]
@chriszs nice catch - thanks for testing! I added your change.
I am unclear if there was also an issue with the readiness check - that one appears to be working for me.

**Comment by chriszs** [comment]
@grugnog I think the readiness check was recording the pod going down when the liveness probe failed.

I'm now having some issues with the Redis queue (pods getting disconnected?), but it's intermittent and I'm not sure what the cause is.

**Comment by chriszs** [comment]
The queue issues seem to be rarer than I initially encountered. Normal stuff that doesn't necessarily warrant a change here.

**Comment by edmondo1984** [comment]
Without any changes, the redash pod for me is failing because redash-server-createdb init contaner is failing `password authentication failed for user "redash"`

**Comment by edmondo1984** [comment]
It is failing for us with authentication problems, either in the init container or in the main pod of redash deployment. 

`sqlalchemy.exc.OperationalError: (psycopg2.OperationalError) FATAL:  password authentication failed for user "redash"`

We are using secrets auto-creation from the chart, so this is kind of surprising

**Comment by grugnog** [comment]
@edmondo1984 could you give some more details on your setup?

As I can't reproduce the issue, we might need some help from you to figure out what is going on - have you been able to confirm (a) if the password is being generated/configured and (b) if it is being set in the redash container environment?

**Comment by zakkg3** [comment]
I had the same problem as @edmondo1984 surprising...

In my redash-server :

```
redash-wc-redash-server-56896d889-9k9nf:/app$ printenv | grep DATABASE_URL
REDASH_DATABASE_URL=postgresql://redash:my-random-passwd@redash-postgresql:5432/redash
```

Logs in redash server:

`OperationalError: (psycopg2.OperationalError) FATAL:  password authentication failed for user "redash"`

So i just attached to the posgresql pod and 

```
psql -U postgres
alter user redash with password 'my-random-passwd';
```

And now in redash-server:

```
./manage.py status
[2019-03-13 09:13:44,051][PID:147][INFO][root] Generating grammar tables from /usr/lib/python2.7/lib2to3/Grammar.txt
[2019-03-13 09:13:44,099][PID:147][INFO][root] Generating grammar tables from /usr/lib/python2.7/lib2to3/PatternGrammar.txt
[2019-03-13 09:13:44,447][PID:147][INFO][root] Latest version: 6.0.0 (newer: True)
{
  "unused_query_results_count": 0, 
  "widgets_count": 0, 
  "redis_used_memory": "2.71M", 
  "dashboards_count": 0, 
  "query_results_count": 0, 
  "manager": {
    "queues": {}
  }, 
  "version": "4.0.0", 
  "queries_count": 0, 
  "workers": []
}

```

I am wondering if bitnamis postgresq did change something recently?


BTW:
 I am using my own chart similar to thisone, with some specific thing we need here, I am just building the connection URLs from environment variables like this:

```
            - name: REDASH_DATABASE_URL
              value: postgresql://$(REDASH_DATABASE_USER):$(REDASH_DATABASE_PASSWORD)@$(REDASH_DATABASE_HOSTNAME):$(REDASH_DATABASE_PORT)/$(REDASH_DATABASE_DB)
```
No need to run  dynamicenv.sh check [this link ](https://kubernetes.io/docs/tasks/inject-data-application/define-environment-variable-container/#using-environment-variables-inside-of-your-config)

Thanks for share!

**Comment by grugnog** [comment]
@zakkg3 if I follow your edits it sounds like you got this working - is that correct? I just retested the latest version on minikube and GKE and it spun up without incident.

Regarding your note "No need to run dynamicenv.sh check this link" you might want to review the older versions of this PR - this was the original approach used. The issue is that we need to use an automatically generated random secret (rather than a default password) for postgresql/redis password inside the connection string environment variables together with other (non-secret) environment variables, which I don't believe is possible without templating the environment variable inside some kind of custom entry point (would be happy to be proven wrong though!). Ultimately I would love to see if upstream could (optionally) support separate variables since this would greatly simplify things.

**Comment by driscollAtBitGo** [comment]
Well, crap.

**Comment by grugnog** [comment]
@unguiculus have you been able to take a look at this?

**Comment by aelbarkani** [comment]
@unguiculus ping

**Comment by maorfr** [comment]
hey @grugnog 

Have you considered hosting this chart and adding it to helm hub?
https://github.com/helm/hub/blob/master/Repositories.md

**Comment by chriszs** [comment]
Happy belated birthday to this pull request.

**Comment by monotek** [comment]
@grugnog 
The DCO needs to be signed to get this merged. 
See: https://github.com/helm/charts/pull/5071/checks?check_run_id=63069193

**Comment by brunowego** [comment]
@grugnog thanks for this! Can you continue accepting the DCO?

**Comment by grugnog** [comment]
DCO is signed (sorry for delay - just got back from vacation!).

**Comment by brunowego** [comment]
@unguiculus ping 🥇

**Comment by maorfr** [comment]
please take a look at https://github.com/helm/helm/pull/6095#issue-301608768

we strongly encourage people to self host chart repos:
https://github.com/helm/hub/blob/master/Repositories.md

you can even do that using GitHub pages:
https://github.com/helm/hub/blob/master/docs/github.md#github-pages-as-a-repository

**Comment by norrs** [comment]
@grugnog : Hey, I would like to request some changes which we did to our vendored chart from this PR inside our organization. 

Please consider adding the following under `env`in the following files: server-deployment.yaml ,  adhocworker-deployment.yaml and scheduledworker-deployment.yaml

```
{{- if .Values.envSecrets.secretKey.existingSecret }}
- name: REDASH_SECRET_KEY
  valueFrom:
    secretKeyRef:
      name: {{ .Values.envSecrets.secretKey.existingSecret | quote }}
      key: secretKey
{{- end }}

```

In values.yaml you can provide the following default values:

```
envSecrets:
  secretKey: {}
    #existingSecret:
```


This way users can use existing secret inside the cluster the secret key that is used in newer versions for redash. See https://version.redash.io/ . This is required for us running with redash7. 
(and production usages want to avoid putting secrets together with deployments, as they either use things like vault etc combined with sealed secret controller)


Here is our internal upgrade transcript we logged when we did our upgrade from redash5 to redash7 below. Maybe it could be useful as a contrib readme or something.





## How to upgrade Redash (database migration)

Example transcript below, in the following example redash5 is old deployment,
and redash7 is new deployment.

### Scale all worker deployments down

...but leave postgresql/redis running:

```
kubectl scale deploy/redash5 --replicas=0
kubectl scale deploy/redash5-adhocworker --replicas=0
kubectl scale deploy/redash5-scheduledworker --replicas=0
```

### Create database dump 

```bash
pg_dump -Fc -U redash redash > /tmp/2019-08-09-upgrade-redash-compressed.backup
```

### Scale old deployment up again

```
kubectl scale deploy/redash5 --replicas=1
kubectl scale deploy/redash5-adhocworker --replicas=2
kubectl scale deploy/redash5-scheduledworker --replicas=2
```

But instruct #general in slack that new edits in queries will not be included
in the upgrade!

### Upgrade chart

Since its not merged to upstream yet ( https://github.com/helm/charts/pull/5071
) , we do the following for vendoring:

```bash
git clone https://github.com/grugnog/charts.git
git checkout redash
(cd stable/redash && helm dependency update)
mkdir /tmp/new-redash
helm package stable/redash --destination /tmp/new-redash
tar -C /tmp/new-redash -zxvf /tmp/new-redash/redash-1.0.0.tgz 
mv /tmp/new-redash/redash ~/zedge/deployments/charts/
```

Ensure we use chart values which refer to correct secrets,
so ensure you inspect the new chart to ensure values are correct.

As of upgrading redash7, redash chart expects specific naming on
redash-postgresql secret to use the { Release.Name }.

### After chart upgrade in a new Helm release

We sadly get a new postgresql instance.

Scale down new deployment while we restore database contents.

```
kubectl scale deploy/redash7 --replicas=0
kubectl scale deploy/redash7-adhocworker --replicas=0
kubectl scale deploy/redash7-scheduledworker --replicas=0
```

### Restore database

Copy dump back in

```bash
kubectl cp ~/tmp/2019-08-09-upgrade-redash-compressed.backup redash7-postgresql-0:/tmp/2019-08-09-upgrade-redash-compressed.backup
```

Enter postgresql pod

```bash
kubectl exec -ti redash7-postgresql-0 /bin/bash -i
```

Restore contents:
```bash
pg_restore -h localhost -p 5432 -U redash -d redash --clean /tmp/2019-08-09-upgrade-redash-compressed.backup
```
(You can ignore the comment about it not being allowed to set a comment on a SQL procedure)
```

### Scale up again

```
kubectl scale deploy/redash7 --replicas=1
kubectl scale deploy/redash7-adhocworker --replicas=2
kubectl scale deploy/redash7-scheduledworker --replicas=2
```

### Run migrations manually

Enter redash7 deployment POD:

```bash
. /config/dynamicenv.sh && /app/bin/docker-entrypoint manage db upgrade
```

The /config/dynamicenv.sh sets the required variables for using `manage db`
commands. (ie the \*\_URL environment variables)


### Success?!

Visit https://redash<version>.example.net  (depends if you use release name in helm set to redash<version>.


## Other useful tricks

*Remember to have a shell which have inherited the /config/dynamicenv.sh !*

### Check the current version and migration history:

```bash
/app/bin/docker-entrypoint manage db current
/app/bin/docker-entrypoint manage db history
```

### See the SQL commands that will be run:

```bash
/app/bin/docker-entrypoint manage db upgrade --sql
```

### Migrate

```bash
/app/bin/docker-entrypoint manage db upgrade
```

## OAuth configuration

Configuration towards Google OAuth is done in GCP Google OAuth project, you
require special access to this project.  When rolling a new version, you
usually want to ensure this one is updated to accept the new javascript
origins, ie https://redash<VERSION>.example.net and a new callback URL before
upgrading to ensure redash versions can run along side each other.

(This section about oauth configuration you could drop for this contrib upgrade readme if you want to include it)

**Comment by zakkg3** [comment]
nono dont stale this, merge it!

**Comment by yzorg** [comment]
> nono dont stale this, merge it!

@zakkg3 This is a long thread, so let me summarize the bad news...

It looks like this PR will **not be merged** because:

> Have you considered hosting this chart and adding it to helm hub?
> https://github.com/helm/hub/blob/master/Repositories.md

from https://github.com/helm/charts/pull/5071#issuecomment-492984825

and a more concise take, from another PR: https://github.com/linkerd/linkerd2/pull/3292#issuecomment-523744466

So I'll open an issue on getredash/redash asking if they'd like to own the helm chart.  In the past they've had [kube docs and scripts](https://github.com/getredash/redash/pull/2178), so I think it is likely they will want to host the helm cart under the getredash org.

If they are not interested we could submit @grugnog's repo to chart hub.  I assume we'll need to cleanup that repo.

**Comment by yzorg** [comment]
Anyone who likes or is following this thread, please vote on the Redash feature discussion: 
https://discuss.redash.io/t/helm-chart-repo/4594

**Comment by arikfr** [comment]
@yzorg @grugnog as per @yzorg's request, I've opened a repo under the `getredash` org on GitHub to host the Helm Chart. I've made both of you as maintainers of this repo (you will need to accept the invite [here](https://github.com/getredash/contrib-helm-chart/invitations)).

Thanks!

**Comment by grugnog** [comment]
@arikfr @yzorg - thanks so much for making this connection and establishing a new repo part of the getredash org.
I think this is the perfect place for the chart to live and I am excited to keep moving this forward there.

I have pushed a rebased version of this PR to that repository and opened a handful of initial tickets. Let's continue work there - PRs are very welcome :D

**Comment by okgolove** [comment]
The end of the big story...

**Pull request closed without merging**
